### DISCUSSION TITLE
Extruder do not go home at pause at layer

### DISCUSSION THREAD

**Discussion by TheOriginalSmitti** [OrcaSlicer/OrcaSlicer#2192] [Q&A]
Hi all
I tried pause at layer which works fine. But the extruder is not moving at home that i can change filament. it is staying at the current position so that is impossible to change the filament.
Can somebody give me a solution?
I am using Ender 3 pro S1.

Many thanks.

**Comment by foreachthing** [comment]
You got to come up with your own pause code.
This is my pause code (printer settings -> machine gcode):
```; START Pause
; Move Head away
G0 X{print_bed_min[0]} Y{print_bed_max[1]} 
G91 ; relative coordinates
G0 Z20 ; move in Z
M0 ; PAUSE
G0 Z-20
G90 ; absolute coordinates
; END Pause
```

this gives _me_ this gcode after exporting:

```
;PAUSE_PRINT
; START Pause
; Move Head away
G0 X0 Y225 
G91 ; relative coordinates
G0 Z20 ; move in Z
M0 ; PAUSE
G0 Z-20
G90 ; absolute coordinates
; END Pause

```

**Reply by TheOriginalSmitti** [reply]
Hi Foreachting
Thank you for your reply. I will try this :)

**Reply by ahmafi** [reply]
I have tried this code, but the printer doesn't print anything after resume. Just moves and nothing flows from nozzle.

**Comment by TheOriginalSmitti** [comment]
Hi
I tried this and it works perfectly with code M25 indeed M0. with M0 I can not exit the pause function on the printer.

**Comment by foreachthing** [comment]
Well, this code was for the Ultimaker 2. I should have mentioned that, sorry.
You'll need to find out how to "exit" pause on your printer.

**Comment by magquintero** [comment]
Greetings, did you find a solution?

**Comment by Zety-1** [comment]
Hello, guys, I wanted to ask if this is the solution or another was found. From already thank you very much. And sorry for the bad English.

**Comment by JPK-MG** [comment]
I just moved over to Orca Slicer and this is the first time I've had to pause, to insert hardware. Surprised that there isn't a background preset on the M25 command that parks the toolhead, holds the temperatures, then restarts at the next coordinate. I saw the code offered here, but Cura has a much nicer implementation and I think it may just be a matter of implementing the pause in Cura, then scooping the code snippet and injecting it into Orca's g-code. It has all the features that make a nice pause and it looks operable as basic g-code. Post back if anyone tries it before I do.

Would love to see this implemented - it just makes sense.

**Comment by Baloo3D** [comment]
Hello everyone, I’m experiencing a similar issue with my printer that uses the MKS Robin Nano board. While the M0 command does not work, the M25 command functions correctly in Cura. However, I’ve been unable to get it to work in OrcaSlicer.
After numerous attempts with different G-code configurations, I discovered that the problem stems from how Cura and OrcaSlicer handle the pause command. In Cura, after the M25 command, the Python code automatically generates additional commands to instruct the extruder on its position before pausing. Unfortunately, OrcaSlicer does not include this code, and I haven’t found a way to restore it.
Some users have suggested using G92 E0, but this does not work with my printer. As a result, after the pause, the extruder stops injecting filament, which affects my prints.
I prefer OrcaSlicer over Cura in nearly every aspect, except for this particular feature. I would greatly appreciate any assistance or solutions that the community can provide. Thank you!

**Comment by Mqrius** [comment]
The better way to do this in OrcaSlicer is probably to check "Manual Filament Change" in Printer Settings > Multimaterial, and then add the pause in the filament change gcode field: Printer Settings > Machine G-Code > Change Filament G-Code, fill in just `M600`.

If you've got a bambulab printer, fill in `M400 U1` instead to get a notification on the touchscreen.

This way allows you to use the different filament functionality of OrcaSlicer natively, allows you to set different settings for the filaments, etc.

Some more discussion here: https://www.reddit.com/r/BambuLab/comments/18fj4ez/why_does_filament_change_in_orca_slicer_not_add/
